### ISSUE TITLE
Remove __all__ use in PICMI and PyPIConGPU

### ISSUE THREAD

**Issue by BrianMarre** [ComputationalRadiationPhysics/picongpu#5014]
PICMI and PyPIConGPU ``__init__.py`` files specify an ``__all__`` variable that is supposed to overwrite the default behavior of ``from picongpu.picmi import *``, but do not actually diverge from the default behavior.

We should therefore not specify them at all.

**Comment by BrianMarre** [comment]
@chillenzer pre-commit removes all import statements from `__init__.py` if we do not include an `__all__` is this intended behaviour?

**Comment by chillenzer** [comment]
Of course, it's not `pre-commit` immediately but `ruff`. These are the pertinent pages: https://docs.astral.sh/ruff/rules/unused-import/ and https://docs.astral.sh/ruff/settings/#lint_ignore-init-module-imports

As the latter is already deprecated, you can decide for yourself if adding a redundant `import ... as ...`, a redundant `__all__` or `#noqa unused-import` (on the file level, of course) is the least bitter pill. I would personally say the former is still slightly easier to maintain but not by as large of a margin as before.
